### DISCUSSION TITLE
I want to  scroll down up to particular page using xpath axes .Is this possible in appium?

### DISCUSSION THREAD

**Discussion by Lakshmanan4233** [appium/appium#18854] [Q&A]
I am tried the all ways .it not work for me. 

Using JavaScript execute,

leave.scrollDownUptoelement(elementByxpath("//*[@text='Applied']/preceding-sibling::*[@text='Test']/preceding-sibling::*[contains(@text,'1') and @index='1']/preceding-sibling::*[@text='Jul 07']"));

also i am trying to get the all elemtns size in the page . But it will return only displayed parts. 

	List<MobileElement> allElements = driver.findElements(MobileBy.id("resourceid"));
		System.out.println(allElements.size());

List allTitleElements = driver.findElementsByAndroidUIAutomator("new UiSelector().resourceId(\"resourceid\")");		
		System.out.println(allTitleElements.size());

Sceanrio:I Want to apply leave then trying to verify the applied leave details in leave list page. In list page lot previously applied and accepeted leave there. So i go for xpath axes . But it doesnot work for me properly.

Please give me possible ways

**Comment by portyu9** [comment]
Hi @Lakshmanan4233 

XPath axes cannot scroll to an element that is not currently present in Android's UI hierarchy.

For a scrollable list, use `UiScrollable` to bring the target into view first:

```java
WebElement element = driver.findElement(
    AppiumBy.androidUIAutomator(
        "new UiScrollable(new UiSelector().scrollable(true))" +
        ".scrollIntoView(new UiSelector().text(\"Applied\"))"
    )
);
```

Then perform your assertions/XPath lookup after that row is visible.

The reason `findElements()` only returns the displayed rows is that controls such as `RecyclerView` recycle their children; off-screen rows may not exist in the accessibility hierarchy yet.

So the fix is: **scroll the list first with `UiScrollable`/`mobile: scroll`, then locate the required row. XPath itself cannot perform the scrolling.**

Please mark this answer as accepted if it helped, thank you.
